.github/workflows/node.js-prod.yml

```diff
@@ -1,75 +1,105 @@
-    needs: [build]
-    runs-on: ubuntu-latest
+        npm test
+      working-directory: backend
 
-    steps:
-    - uses: actions/checkout@v3
-    - name: Copy file via scp
-      uses: appleboy/scp-action@master
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
       with:
-        host: ${{ secrets.SSH_HOST }}
-        username: ${{ secrets.SSH_USER }}
-        port: ${{ secrets.SSH_PORT }}
-        password: ${{ secrets.SSH_PASS }}
-        source: "./backend"
-        target: "/var/warehouse/makerdao-mips"
+        aws-access-key-id: ${{ secrets.PROD_AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.PROD_AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ env.REGION }}
+
+    - name: Login to AWS ECR
+      id: login-ecr
+      uses: aws-actions/amazon-ecr-login@v2
+
+    - name: Extract commit hash
+      id: vars
+      if: ${{ !contains(github.event.head_commit.message , '[skip build]') }}
+      shell: bash
+      run: |
+        echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
 
-    - name: Executing remote command
-      uses: appleboy/ssh-action@master
+    - name: Build, tag, and push image to ECR
+      id: build-image
+      if: ${{ !contains(github.event.head_commit.message , '[skip build]') }}
+      working-directory: backend
       env:
+        SHA_TAG: ${{ steps.vars.outputs.sha_short }}
+        LATEST_TAG: latest
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
         REQUEST_GITHUB_URL_API_ENDPOINT: ${{ secrets.REQUEST_GITHUB_URL_API_ENDPOINT }}
-        MIP_GITHUB_REPOSITORY: ${{ secrets.MIP_GITHUB_REPOSITORY_PROD }}
-        MIP_GITHUB_REPOSITORY_OWNER: ${{ secrets.MIP_GITHUB_REPOSITORY_OWNER_PROD }}
         GIT_ACCESS_API_TOKEN: ${{ secrets.GIT_ACCESS_API_TOKEN }}
+        MIP_GITHUB_REPOSITORY: ${{ secrets.MIP_GITHUB_REPOSITORY_DEV }}
+        MIP_GITHUB_REPOSITORY_OWNER: ${{ secrets.MIP_GITHUB_REPOSITORY_OWNER_DEV }}
+      run: |
+        # Build Docker containers and push them to ECR ${{ env.AWS_ECR_NAME }}
+        docker pull $ECR_REGISTRY/$AWS_ECR_NAME:$LATEST_TAG || true
+        docker build -t $AWS_ECR_NAME \
+                     -t $ECR_REGISTRY/$AWS_ECR_NAME:$SHA_TAG \
+                     -t $ECR_REGISTRY/$AWS_ECR_NAME:$LATEST_TAG \
+                     -t $ECR_REGISTRY/$AWS_ECR_NAME:$ENVIRONMENT_TAG \
+                     --build-arg REQUEST_GITHUB_URL_API_ENDPOINT=$REQUEST_GITHUB_URL_API_ENDPOINT \
+                     --build-arg GIT_ACCESS_API_TOKEN=$GIT_ACCESS_API_TOKEN \
+                     -f Dockerfile \
+                     .
+
+        docker push $ECR_REGISTRY/$AWS_ECR_NAME --all-tags
+
+    - name: Replace variables in the Helm values file
+      id: replace-vars
+      if: ${{ !contains(github.event.head_commit.message , '[skip deploy]') }}
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+      run: |
+        sed -i 's/${ECR_REGISTRY}/'$ECR_REGISTRY/ $HELM_FILE
+
```

Comment:
Not needed since you're passing a value for `image.repository` that takes precedence.
```suggestion
```